phpseclib/Common/Functions/Objects.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+/**
+ * Common Object Functions
+ *
+ * PHP version 5
+ *
+ * @category  Common
+ * @package   Functions\Objects
+ * @author    Jim Wigginton <terrafrost@php.net>
+ * @copyright 2016 Jim Wigginton
+ * @license   http://www.opensource.org/licenses/mit-license.html  MIT License
+ * @link      http://phpseclib.sourceforge.net
+ */
+
+namespace phpseclib\Common\Functions;
+
+/**
+ * Common Object Functions
+ *
+ * @package Functions\Objects
+ * @author  Jim Wigginton <terrafrost@php.net>
+ */
+abstract class Objects
```

Comment:
But that said, it is not immediately obvious to me how to best deal with the problem. Maybe when I'm neck deep in SSH2.php I'll come up with a better idea but right now I'm almost finished with my DSA implementation. I want to do some more BigInteger changes and then do ECDSA and EdDSA. And then more ASN1.php changes and then X509.php changes, etc.

But as for why it's not immediately obvious to me...

SCP.php calls `get_channel_packet` in SSH2.php. That is NOT  a method people should normally be calling. That's why `get_channel_packet` is declared private. BUT SCP.php is not a regular file.

System/SSH/Agent.php calls `get_channel_packet` as well. ie. SSH2.php isn't setting SSH forwarding up - System/SSH/Agent.php is.

Now, maybe all of this will be changed when I get around to making the big changes for SSH2.php but until then...  these are functions that /regular/ applications ought not be calling but that some other classes within phpseclib /do/ have a current need to call. In C++ this is where you'd use `friend` but PHP doesn't have anything similar to that.

So what would you recommend in that scenario?